sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/UnusedPrivateMemberTest.cs

```diff
@@ -714,6 +714,46 @@ public class NewClass
 ", new UnusedPrivateMember());
         }
 
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Unity3D_Ignored()
+        {
+            Verifier.VerifyCSharpAnalyzer(@"
+// https://github.com/SonarSource/sonar-csharp/issues/159
+public class UnityMessages1 : UnityEngine.MonoBehaviour
+{
+    private void SomeMethod(bool hasFocus) { } // Compliant
+}
+
+public class UnityMessages2 : UnityEngine.ScriptableObject
+{
+    private void SomeMethod(bool hasFocus) { } // Compliant
+}
+
+public class UnityMessages3 : UnityEditor.AssetPostprocessor
+{
+    private void SomeMethod(bool hasFocus) { } // Compliant
+}
+
+public class UnityMessages4 : UnityEditor.AssetModificationProcessor
+{
+    private void SomeMethod(bool hasFocus) { } // Compliant
+}
+
+// Unity3D does not seem to to available as a nuget package and we cannot use the original classes
```

Comment:
FYI I don't think it not being available as a NuGet package is a problem. Since we don't check the assembly the class is defined in, I don't think there's much value in referencing the real package over defining our own classes with the expected names anyway - it just adds bloat to the tests, and doesn't scale well.